scripts/syncVersions.ts

```diff
@@ -0,0 +1,107 @@
+    readonly devDependencies?: Record<string, string>;
+    readonly peerDependencies?: Record<string, string>;
+}
+
+type Dependencies = keyof Pick<PackageJson, 'dependencies' | 'devDependencies' | 'peerDependencies'>;
+
+async function loadPackageJson(packageJsonPath: string): Promise<PackageJson> {
+    const error = new Error(`${packageJsonPath} does not seem to be a valid package.json file`);
+
+    let packageJson: Partial<PackageJson>;
+    try {
+        packageJson = JSON.parse(await fs.promises.readFile(packageJsonPath, 'utf-8')) as Partial<PackageJson>;
+    } catch {
+        throw error;
+    }
+
+    if (!packageJson.name || !packageJson.version) {
+        throw error;
+    }
+
+    return packageJson as PackageJson;
+}
+
+async function savePackageJson(packageJsonPath: string, packageJson: PackageJson) {
+    return await fs.promises.writeFile(packageJsonPath, JSON.stringify(packageJson, null, '    ') + '\n');
+}
+
+function getWorkspacePackageJsonPaths(packageJson: PackageJson) {
+    return packageJson.workspaces?.map((workspaceDir) => path.join(rootDir, workspaceDir, 'package.json')) ?? [];
+}
+
+function updateDependency(packageJson: PackageJson, type: Dependencies, name: string, newVersion: string) {
+    const deps = packageJson[type];
+    if (!deps) {
+        return false;
+    }
+
+    const currentVersion = deps[name];
+    if (!currentVersion) {
+        return false;
+    }
+
+    if (currentVersion !== newVersion) {
+        deps[name] = newVersion;
+        console.log(`[${packageJson.name}] - updated ${name} from ${currentVersion} to ${newVersion} in ${type}`);
+
+        return true;
+    }
+
+    return false;
+}
+
+function updateVersions(packageJson: PackageJson, currentVersions: Record<string, string>) {
+    let updated = false;
+
+    for (const [name, version] of Object.entries(currentVersions)) {
+        const newVersion = `^${version}`;
+
+        updated = updateDependency(packageJson, 'dependencies', name, newVersion) || updated;
+        updated = updateDependency(packageJson, 'devDependencies', name, newVersion) || updated;
+        updated = updateDependency(packageJson, 'peerDependencies', name, newVersion) || updated;
+    }
+
+    if (!updated) {
+        console.log(`[${packageJson.name}] - no changes`);
+    }
+
+    return packageJson;
+}
+
+(async () => {
+    const rootPackageJson = await loadPackageJson(path.join(rootDir, 'package.json'));
+    const workspacePackageJsonPaths = getWorkspacePackageJsonPaths(rootPackageJson);
+    const workspacePackageJsons = await Promise.all(workspacePackageJsonPaths.map(loadPackageJson));
+
+    const args = process.argv.slice(2);
+    const packageJsonPathsToSync = args.length ? args : workspacePackageJsonPaths;
+
+    const currentVersions = workspacePackageJsons.reduce((obj, pj) => {
+        obj[pj.name] = pj.version;
+        return obj;
+    }, {} as Record<string, string>);
+
+    for (const packageJsonPath of packageJsonPathsToSync) {
```

Comment:
can we run `npm install` in the end if there is any change? 

we're using `npm ci` which requires from us to keep our package-lock.json up to date. We can also automate this by using this tool. 

WDYT?

Reply:
I wouldn't do that. Even if we decide that, it should be a part of scripts in package.json: 
```
ts-node ./scripts/syncVersions.ts && npm install
```